src/features/GetCommands.ts

```diff
@@ -0,0 +1,104 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+import * as vscode from "vscode";
+import { LanguageClient, RequestType } from "vscode-languageclient";
+import { IFeature } from "../feature";
+
+export const GetAllCommandsRequestType = new RequestType<any, any, void, void>("powerShell/getAllCommands");
+
+export class GetCommandsFeature implements IFeature {
+    private command: vscode.Disposable;
+    private languageClient: LanguageClient;
+    private commandsExplorerProvider: CommandsExplorerProvider;
+
+    constructor() {
+        this.command = vscode.commands.registerCommand("PowerShell.RefreshCommandsExplorer", () => {
+            if (this.languageClient === undefined) {
+                // TODO: Log error message
+                return;
+            }
+            this.languageClient.sendRequest(GetAllCommandsRequestType, "").then((result) => {
+                this.commandsExplorerProvider.powerShellCommands = result.map(toCommand);
+                this.commandsExplorerProvider.refresh();
+            });
+        });
+        this.commandsExplorerProvider = new CommandsExplorerProvider();
+        vscode.window.registerTreeDataProvider("PowerShellCommands", this.commandsExplorerProvider);
+        vscode.commands.registerCommand("PowerShell.InsertCommand", (item) => {
+            const editor = vscode.window.activeTextEditor;
+            const sls = editor.selection.start;
+            const sle = editor.selection.end;
+            const range = new vscode.Range(sls.line, sls.character, sle.line, sle.character);
+            editor.edit((editBuilder) => {
+                editBuilder.replace(range, item.Name);
+            });
+        });
+    }
+
+    public dispose() {
+        this.command.dispose();
+    }
+
+    public setLanguageClient(languageclient: LanguageClient) {
```

Comment:
Where does this function get called?

Reply:
If you're referring to setLanguageClient, I believe this is called by vscode itself, I'd have to look at the API docs, but I do know it gets called (or at least a breakpoint in it triggers)

Reply:
This method is part of the PowerShell extension infrastructure defined in feature.ts.   When a PowerShell session is started (in session.ts), it starts the language client.  When the language client is ready, the SessionManager walks through its array of features and updates each with the new language client.

Reply:
As @rkeithhill said, it is called by the SessionManager. It doesn't work well (at all) without the function.